### PR TITLE
EHPR-95 | Zero install Docker

### PR DESCRIPTION
**Pull request by fwpushan**
- [x] Fix web build issue
- [x] Standalone container to watch common changes 
- [x] Attach common changes with api and web container.
- [x] Create uniform cmd interface for all containers. 
- [x] Fix some file copy issue related to aws api build zip

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -42,12 +41,11 @@ services:
     build:
       context: ./
       dockerfile: Dockerfile
-    command: 'yarn watch:web'
+    command: 'yarn workspace @ehpr/web start'
```
Comment (arranfw) [inline docker-compose.yml:44]: I'm not a huge fan of the inconsistency in commands here

**Review by arranfw — COMMENTED** [review]
It would be nice to have things a bit more organized, I liked the idea of having a container running yarn watch for the common package, it doesn't totally make sense why api is watching common and web isn't

File: Dockerfile [hunk 2]
```diff
@@ -2,16 +2,10 @@ FROM node:16-alpine
 
 # RUN yarn set version berry
 # Copying repo resources
-COPY ./package.json ./
-COPY ./tsconfig.json ./
 COPY ./packages ./packages
-COPY ./nodemon.json ./
-COPY ./.eslintrc.js ./
 COPY ./apps ./apps
```
Comment (sunghwan-park-fw) [inline Dockerfile:6]: I am not sure if this copy command is consistent with the volume mappings in docker-compose.yml.

File: docker-compose.yml (hunk 1 above)
Comment (fwpushan) [reply docker-compose.yml]: FIXED

Comment (fwpushan) [reply docker-compose.yml]: Fixed

**Comment by fwpushan** [comment]
> It would be nice to have things a bit more organized, I liked the idea of having a container running yarn watch for the common package, it doesn't totally make sense why api is watching common and web isn't

@arranfw Implemented a standalone watch container for common and attach them with api and web. Everything is running very smooth in my local host.

File: Dockerfile (hunk 2 above)
Comment (fwpushan) [reply Dockerfile:6]: COPY cmd copy asset to image (static binding) build phase.
VOLUME mounting is attaching local file system with running container (Dynamic).

**Review by dbayly-freshworks — APPROVED** [review]
I just pulled this down locally and it works great. Good job!

File: Dockerfile [hunk 3]
```diff
@@ -2,16 +2,10 @@ FROM node:16-alpine
 
 # RUN yarn set version berry
 # Copying repo resources
-COPY ./package.json ./
-COPY ./tsconfig.json ./
 COPY ./packages ./packages
```
Comment (hitankar-freshworks) [inline Dockerfile:5]: Do we need to even copy packages and apps if we are mounting them?

File: packages/common/src/hello-world.ts [hunk 4]
```diff
@@ -1 +1 @@
-export const helloWorld = () => 'Hello World EHPR!!! :)';
+export const helloWorld = () => 'Hello World EHPR!!!';
```
Comment (hitankar-freshworks) [inline packages/common/src/hello-world.ts:1]: we can remove this file.

File: Dockerfile (hunk 2 above)
Comment (arranfw) [reply Dockerfile:6]: I think we shouldn't need to copy if we mount directory as a volume

**Comment by hitankar-freshworks** [comment]
LGTM, one comment

**Review by hitankar-freshworks — APPROVED** [review]

File: apps/web/.babelrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["next/babel"]
```
Comment (arranfw) [inline apps/web/.babelrc:2]: Why did this need to be added?

Comment (sunghwan-park-fw) [reply apps/web/.babelrc:2]: maybe https://nextjs.org/docs/messages/failed-loading-swc

Comment (fwpushan) [reply apps/web/.babelrc:2]: web building is failing in docker env.

Comment (sunghwan-park-fw) [reply apps/web/.babelrc:2]: I guess it is because SWC is a binary module specific to the platform and we don't install node modules in the container.

**Pull request merged**